src/middleware/routeHandler.js

```diff
@@ -8,16 +8,39 @@ const { default: GithubSessionData } = require("@classes/GithubSessionData")
 const { lock, unlock } = require("@utils/mutex-utils")
 const { getCommitAndTreeSha, revertCommit } = require("@utils/utils.js")
 
+const { ConflictError } = require("@root/errors/ConflictError")
 const GitFileSystemError = require("@root/errors/GitFileSystemError")
 const {
   default: GitFileSystemService,
 } = require("@services/db/GitFileSystemService")
 
+const logger = require("@logger/logger").default
+
 const WHITELISTED_GIT_SERVICE_REPOS = config.get(
   "featureFlags.ggsWhitelistedRepos"
 )
 const BRANCH_REF = config.get("github.branchRef")
 
+const gitFileSystemService = new GitFileSystemService(new SimpleGit())
+
+const isRepoWhitelisted = (siteName) =>
```

Comment:
hmm could we shift this out into a separate place since this logic is exactly the same in `ReposService.ts` ?

Reply:
used to have this is a separate utils called ggs-utils.ts, but we removed it because we didn't want any other file to be aware of this special GGS logic apart from RepoService

However, for this routeHandler, I think we don't really have a choice? Unless we inject RepoService as a dependency into routeHandler instead of GitFIleSystemService. So RepoService will be the middleman for the isGitLocked() method. That will be a possible approach

Reply:
I opted for attaching `GitFileSystemService` because it has no dependencies, whereas having `RepoService` instead would mean having to initialise a separate `RepoService` (and its dependencies) inside the route handler!

Reply:
I am confused here :( we abstract this `isRepoWhitelisted` method, there wouldnt be any dep on `RepoService` no?